library/src/main/java/com/bumptech/glide/load/engine/cache/ExternalCacheDiskCacheFactory.java

```diff
@@ -0,0 +1,30 @@
+package com.bumptech.glide.load.engine.cache;
+
+import android.content.Context;
+
+import java.io.File;
+
+/**
+ * Creates an {@link com.bumptech.glide.disklrucache.DiskLruCache} based disk cache in the external disk cache
+ * directory.
+ * Images can be read by everyone when using external disk cache.
+ */
+public final class ExternalCacheDiskCacheFactory extends DefaultDiskCacheFactory {
+
+    public ExternalCacheDiskCacheFactory(Context context, int diskCacheSize) {
+        this(context, null /*diskCacheName*/, diskCacheSize);
+    }
+
+    public ExternalCacheDiskCacheFactory(Context context, String diskCacheName, int diskCacheSize) {
+        super(getExternalCacheFolder(context), diskCacheName, diskCacheSize);
+    }
+
+    private static String getExternalCacheFolder(Context context) {
+        File cacheDir = context.getExternalCacheDir();
```

Comment:
as @TWiStErRob mentioned, this is unfortunately not safe because getCacheDir and getExternalCacheDir both cause strict mode violations.